spec/action_spec.rb

```diff
@@ -8,6 +8,9 @@ def run(action)
 
   before :each do
     @machine.stub(:id).and_return(EXISTING_UUID)
+    # Vagrant has some pretty buggy multi threading and their conditions
+    # check can fail if the wait_for_ready method returns right away
+    @machine.communicate.stub(:wait_for_ready) { sleep(1) }
```

Comment:
> Sorry for all of the commits, the tests were in a pretty broken state when I got to them.
>
> The commit message has details, but we have to make sure that our mock takes a little while to "connect" because Vagrant has some really terrible multithreaded code here: https://github.com/mitchellh/vagrant/blob/master/lib/vagrant/action/builtin/wait_for_communicator.rb#L16

Should this resolve #181? I was getting all tests passing locally this morning, but some Travis builds were failing on just that one test, and re-running the build seemed to fix it. I created the issue so I could remember to look into it when I have more time, but is it already taken care of here?